autoretrieve/engine.go

```diff
@@ -0,0 +1,524 @@
+package autoretrieve
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"sync"
+
+	"github.com/filecoin-project/go-legs"
+	"github.com/filecoin-project/go-legs/dtsync"
+	"github.com/filecoin-project/go-legs/httpsync"
+	provider "github.com/filecoin-project/index-provider"
+	"github.com/filecoin-project/index-provider/engine/chunker"
+	"github.com/filecoin-project/index-provider/metadata"
+	"github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	dsn "github.com/ipfs/go-datastore/namespace"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+const (
+	keyToCidMapPrefix      = "map/keyCid/"
+	cidToKeyMapPrefix      = "map/cidKey/"
+	keyToMetadataMapPrefix = "map/keyMD/"
+	latestAdvKey           = "sync/adv/"
+	linksCachePath         = "/cache/links"
```

Comment:
are these namespaces not too generic? should we add the package name along `autoretrieve//cache/links` ?

Reply:
This is not my code, I stole it from https://github.com/filecoin-project/index-provider/blob/dfc4fa4ad02a4f0494b67f7f6c15581c434ef89b/engine/engine.go#L25-L31.